Tests/Get-ProGetUniversalPackage.Tests.ps1

```diff
@@ -2,227 +2,219 @@
-        $params['GroupName'] = $InGroup
-    }
+        $params = @{ }
+        if( $InGroup )
+        {
+            $params['GroupName'] = $InGroup
+        }
 
-    $script:upackFile = New-ProGetUniversalPackage -OutFile (Join-Path -Path $TestDrive.FullName -ChildPath ('{0}{1}.zip' -f $InGroup,$Named)) -Version '0.0.0' -Name $Named @params
-    Publish-ProGetUniversalPackage -Session $session -FeedName $feedName -PackagePath $upackFile
-}
+        $filepath = Join-Path -Path $TestDrive -ChildPath ('package.{0}.upack' -f [IO.Path]::GetRandomFileName())
+        $upackFile = New-ProGetUniversalPackage -OutFile $filepath -Version '0.0.0' -Name $Named @params
+        Publish-ProGetUniversalPackage -Session $session -FeedName $feedName -PackagePath $upackFile
+    }
 
-function Init
-{
-    $script:result = $null
-    $script:upackFile = $null
-    GivenFeed $feedName
-}
+    function Init
+    {
+        $script:result = $null
+        $script:upackFile = $null
+        GivenFeed $feedName
+    }
 
-function ThenError
-{
-    param(
-        $Matches
-    )
+    function ThenError
+    {
+        param(
+            $Matches
+        )
 
-    It ('should write error') {
         $Global:Error | Should -Match $Matches
     }
-}
 
-function ThenNoErrors
-{
-    param(
-    )
+    function ThenNoErrors
+    {
+        param(
+        )
 
-    It ('should write no errors') {
         $Global:Error | Should -BeNullOrEmpty
     }
-}
 
-function ThenNothingReturned
-{
-    It ('should return nothing') {
+    function ThenNothingReturned
+    {
         $result | Should -BeNullOrEmpty
     }
-}
 
-function ThenPackageReturned
-{
-    param(
-        [string[]]
-        $Named,
-
-        $InGroup = ''
-    )
+    function ThenPackageReturned
+    {
+        param(
+            [string[]]
+            $Named,
 
-    It ('should return package') {
+            $InGroup = ''
+        )
+        $result = $script:result
         $result | Should -HaveCount $Named.Count
         foreach( $name in $Named )
         {
             $result | Where-Object { $_.name -eq $name } | Should -not -BeNullOrEmpty
             $result | Where-Object { $_.group -eq $InGroup } | Should -HaveCount $Named.Count
         }
     }
+
+    function WhenGettingPackage
+    {
+        [CmdletBinding()]
+        param(
+            $Named,
+            $InGroup
+        )
+
+        $params = @{ }
+        if( $PSBoundParameters.ContainsKey('Named') )
+        {
+            $params['Name'] = $Named
+        }
+        if( $PSBoundParameters.ContainsKey('InGroup') )
+        {
+            $params['GroupName'] = $InGroup
+        }
+        $Global:Error.Clear()
+        $script:result = Get-ProGetUniversalPackage -Session $session -FeedName $feedName @params
+    }
 }
 
-function WhenGettingPackage
-{
-    [CmdletBinding()]
-    param(
-        $Named,
-        $InGroup
-    )
+Describe 'Get-ProGetUniversalPackage when package doesn''t exist' {
+    BeforeEach {
+        Init

```

Comment:
Since there will only be one `Describe` block, there will only be one `BeforeEach` block, so move the `Init` function's code into the single `BeforeEach` block and delete the `Init` function.